Emitter-related tweaks and changes

atlantiscze opened the pull request:
This PR is slightly inspired by the staff meeting. It keeps emitters usable, but it prevents them from being way too overpowered which is what they are now (capable of punching through r-wall within 8 pulses, or soloing a blob). Instead of reflective walls (which i still find kind of silly) i added reinforcement for materials. In other words, plasteel reinforced walls take 10x less damage from projectiles on any kind. Steel walls take 5x less damage, et cetera. It is all material datum based. Rough amount of shots required to burn through a r-wall increased from 8 to a little over 50. That means it still works, just not within ten seconds. A r-wall will hold few minutes.

The same goes for blobs. They are generally stronger, and take much less damage from lasers in general. An emitter is still useful. It keeps the blob suppressed a bit, but it shouldn't be able to solo the core. Multiple emitters should still work quite reliably against a single-core blob, but i find that acceptable.
- Emitters now use 100kW, instead of 30kW. That means it is only possible to run them with a regular PACMAN if you overload it (which has it's risks if not watched closely). Also makes it a little bit more expensive to run one off the grid as usual.
- Blobs are now considerably more resistant to emitters (and lasers in general). Emitters are still useful to keep the blob under stress. They will cause damage, and may actually help you damage the core a bit, but they shouldn't be capable of outright destroying the core. Multiple emitters shooting from all directions would be required for that, or an engineer with welder or other weapon.
- BONUS: Blobs now have a 2.5% chance to grow a secondary core instead of regular blob segment. This secondary core is weaker in terms of growth - it will only grow 4 tiles away, and is generally easier to kill, but it is still a nuisance.
- BONUS2: I recycled two blob icon states and used them for the core. The blob core sprite will now change to reflect how damaged it is.
- Materials now have a new variable that indicates how good are they as reinforcement material. This is used by walls. If a wall is reinforced with a material, any projectile damage gets divided by this reinforcement potential. Practical effect is that emitters and other handheld weaponry are much worse against r-walls now. You need over 50 shots to bring one down. This shouldn't affect regular non-reinforced walls.
- Note: Steel is a decent reinforcement material with 80% reduction. Plasteel is even better with 90% reduction. Titanium is a little bit over 92%, however it has worse overall health and heat resistance as compared to plasteel. Most common materials have values around 50%, with some weaker ones (wood) having outright 0% reduction.
- Fix: Glass materials now have melting point set. That means a borosilicate glass plated wall won't melt at room temperature.

atlantiscze committed:
Emitter-related tweaks and changes
- Emitters now use 100kW, instead of 30kW. That means it is only possible to run them with a regular PACMAN if you overload it (which has it's risks if not watched closely). Also makes it a little bit more expensive to run one off the grid as usual.
- Blobs are now considerably more resistant to emitters. Emitters are still useful to keep the blob under stress. They will cause damage, and may actually help you damage the core a bit, but they shouldn't be capable of outright destroying the core. Multiple emitters shooting from all directions would be required for that, or an engineer with welder or other weapon.
- BONUS: Blobs now have a 2.5% chance to grow a secondary core instead of regular blob segment. This secondary core is weaker in terms of growth - it will only grow 4 tiles away, and is generally easier to kill, but it is still a nuisance.
- Materials now have a new variable that indicates how good are they as reinforcement material. This is used by walls. If a wall is reinforced with a material, any projectile damage gets divided by this reinforcement potential. Practical effect is that emitters and other handheld weaponry are much worse against r-walls now. You need over 50 shots to bring one down. This shouldn't affect regular non-reinforced walls.
- Note: Steel is a decent reinforcement material with 80% reduction. Plasteel is even better with 90% reduction. Titanium is a little bit over 92%, however it has worse overall health and heat resistance as compared to plasteel. Most common materials have values around 50%, with some weaker ones (wood) having outright 0% reduction.
- Fix: Glass materials now have melting point set. That means a borosilicate glass plated wall won't melt at room temperature.

meyar commented:
As one of the admins who was raising concerns about emitters just ruining an AI's shit, I find this acceptable.

Kasuobes commented:
How do you even make titanium?

atlantiscze commented:
I believe it can not actually be made, but it can exist as a wall material

Datraen commented on code/modules/blob/blob.dm:
```diff
@@ -50,7 +54,7 @@
 
 /obj/effect/blob/proc/regen()
-	health = min(health + 1, maxHealth)
+	health = min(health + regen_rate, maxHealth)
```
In addition to having the resistances buffed and the emitter nerfed, making the blob regenerate at 5x the normal speed?

Datraen replied:
Worth also including that the core regens at 2x the previous speed.

BlueNexus replied:
I don't see the issue. An emitter could still eventually kill the blob on its own if it wasn't for the higher regen rate.

Datraen replied:
I spotted the change while going through, as it was not listed.

Having a large draw required to operate the emitters does act as a significant nerf to them, but this would also make killing blobs in general a much more difficult task, and should be noted as a general blob change, and not just an emitter-related change.

atlantiscze replied:
The power usage is only really noticeable when either:
A) Running on solars, or in general without good power supply. Engine usually outputs at least 1MW, which is more than enough for even multiple emitters
B) Running on a PACMAN. Now you have to set it to 5/5, which will eventually cause it to overheat. In other words, you have to watch it closely and give it an occassional cooldown every here and then.

Regen rate is only really noticeable with the core and to a limited degree with the shield blobs (those dense ones around the core). Regular blobs can still be easily killed with few hits with a welding tool - this wasn't changed.

Kelenius commented:
I want to say "what the fuck are you doing" but I think I'll just wait until a couple of blobs pop up on the station after this change and everyone hates you.

Raptor1628 commented:
In a more tactful way: tweaking blob and wall effectiveness up, and emitter effectiveness down might be too much off the bat. Balance tends to work best in small steps. But I'll wait to see how this works.

comma commented:
Wouldn't this change make walls effectively impenetrable to bullets? 10x more damage needed for rwall is quite a step up.

BlueNexus commented:
It doesn't really make sense that bullets can destroy walls, anyway. These are solid metal reinforced walls. I support these changes, regardless.

comma commented:
They are hardly solid. There's grider undearneath, with metal plating on top of it. They are not solid 2x2x2 slabs of metal.
'Makes sense' is about as heavy argument as 'realistic'. Currently shooting a lot carries some risk as you can breach your surroundings, especially with assault rifles and such. This change removes that.

atlantiscze commented:
The change only affects reinforced walls - regular walls are still pretty fragile. A plasteel reinforced wall is probably closer to a solid slab of metal. On the other hand, regular steel non-reinforced wall is probably a thin steel sheet mounted on a girder.

I personally found it very silly to be able to break into AI core simply by unloading one or two energy guns into a reinforced wall. You don't even need an emitter. An energy gun, recharger, or advanced energy gun (or captain's laser gun) are more than enough. Simply shoot through the wall, and when you break through continue shooting until the AI dies, without having to risk turret expousure.

BlueNexus commented:
Status on this?

PsiOmegaDelta commented on code/modules/materials/materials.dm:
```diff
@@ -90,6 +90,7 @@ var/list/name_to_material
 	var/radioactivity            // Radiation var. Used in wall and object processing to irradiate surroundings.
 	var/ignition_point           // K, point at which the material catches on fire.
 	var/melting_point = 1800     // K, walls will take damage if they're next to a fire hotter than this
+	var/projectile_armor = 2	 // When a wall is reinforced by this material, projectile damage to the wall is divided by this. Applies to all projectile damage types!
```
Think I'd rather see damage reduction be general (possibly per damage type) rather than tailored for all possible sources as they're considered.

atlantiscze committed:
Splits burn/brute damage types for wall armors.
- These values match for most materials, but few of them have different fire/brute resistances (for example, borosilicate glass has very good fire resistance, but quite poor brute resistance)

atlantiscze commented:
The brute/burn damage type armors are now handled separately.

PsiOmegaDelta reviewed (approved):
..damage handling really needs to be generalized. As is it really leads to a silly amount of var definitions.
